c++/src/CMakeLists.txt

```diff
@@ -218,8 +218,8 @@ target_include_directories (orc
   INTERFACE
     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
   PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/c++/include>
-    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/c++/include>
+    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/c++/include>
+    $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/c++/include>
```

Comment:
This is not related to this PR but... In general, build directory -> source directory order is better.